ChargingStation.Backend/Infrastructure/ChargingStation.Infrastructure/AzureTableStorage/AzureTableStorageManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Azure.Data.Tables;
+
+namespace ChargingStation.Infrastructure.AzureTableStorage
+{
+    public class AzureTableStorageManager<T> where T : BaseTableStorageEntity
+    {
+        private readonly TableServiceClient _tableServiceClient;
+        private readonly string _tableName;
+
+        public AzureTableStorageManager(string connectionString, string tableName)
+        {
+            _tableServiceClient = new TableServiceClient(connectionString);
+            _tableName = tableName;
+        }
+
+        public async Task AddEntityAsync(T entity)
+        {
+            var tableClient = GetTableClient();
+            await tableClient.AddEntityAsync(entity);
+        }
+
+        public async Task<T> GetEntityAsync(string partitionKey, string rowKey)
+        {
+            var tableClient = GetTableClient();
+            return await tableClient.GetEntityAsync<T>(partitionKey, rowKey);
+        }
+
+        public async Task<IEnumerable<T>> GetEntitiesByPartitionKeyAsync(string partitionKey)
+        {
+            var tableClient = GetTableClient();
+            return await tableClient.QueryAsync<T>(filter: $"PartitionKey eq '{partitionKey}'").ToListAsync();
+        }
+
+        public async Task<IEnumerable<T>> GetAllEntitiesAsync()
+        {
+            var tableClient = GetTableClient();
+            return await tableClient.QueryAsync<T>().ToListAsync();
+        }
+
+        public async Task UpdateEntityAsync(T entity)
```

Comment:
Rename to Upsert, because in this method "Upsert" method is being called, that creates or updates entity